include/taco/util/env.h

```diff
@@ -21,23 +23,47 @@ inline std::string getFromEnv(std::string flag, std::string dflt) {
 }
 
 inline std::string getTmpdir() {
-  // use POSIX logic for finding a temp dir
-  auto tmpdir = getFromEnv("TMPDIR", "/tmp/");
-  
-  // if the directory does not have a trailing slash, add one
-  if (tmpdir.back() != '/') {
-    tmpdir += '/';
-  }
-  
-  // ensure it is an absolute path
-   taco_uassert(tmpdir.front() == '/') <<
-    "The TMPDIR environment variable must be an absolute path";
+  if (cachedtmpdir == ""){
+    // use posix logic for finding a temp dir
+    auto tmpdir = getFromEnv("TMPDIR", "/tmp/");
+
+    // if the directory does not have a trailing slash, add one
+    if (tmpdir.back() != '/') {
+      tmpdir += '/';
+    }
+
+    // ensure it is an absolute path
+     taco_uassert(tmpdir.front() == '/') <<
+      "The TMPDIR environment variable must be an absolute path";
+
+    taco_uassert(access(tmpdir.c_str(), W_OK) == 0) <<
+      "Unable to write to temporary directory for code generation. "
+      "Please set the environment variable TMPDIR to somewhere writable";
 
-  taco_uassert(access(tmpdir.c_str(), W_OK) == 0) <<
-    "Unable to write to temporary directory for code generation. "
-    "Please set the environment variable TMPDIR to somewhere writable";
+    // ensure that we use a taco tmpdir unique to this process.
+    auto tacotmpdirtemplate = tmpdir + "taco_tmp_XXXXXX";
+    char *ctacotmpdirtemplate = new char[tacotmpdirtemplate.length() + 1];
+    std::strcpy(ctacotmpdirtemplate, tacotmpdirtemplate.c_str());
```

Comment:
Is this copy to work around `const`-ness?

Reply:
yes. I'm not sure if c_str is actually const, but the header says not to modify the return c_string.